pkg/interface/chat/src/js/components/chat.js

```diff
@@ -78,12 +83,22 @@ export class ChatScreen extends Component {
     } else if (
       props.envelopes.length >= prevProps.envelopes.length + 10
     ) {
+      if(navigator.userAgent.includes('Firefox')) {
+        // new messages came in, restore FF scroll pos
+        this.recalculateScrollTop();
+      }
       this.hasAskedForMessages = false;
       if (prevProps.envelopes.length <= 20) {
         this.setState({ scrollLocked: false }, () => {
           this.scrollToBottom();
         })
       }
+    } else if (
+      navigator.userAgent.includes("Firefox") &&
+        props.length !== prevProps.length &&
+        state.scrollBottom
+    ) {
+      this.scrollToBottom();
```

Comment:
Ha, if this is here perhaps remove what I did on L91:L94 here — Firefox specific scrolling to bottom should be definitely cordoned off in a user agent flag so we can organise and remove it when we can properly simplify this.